codalab/apps/web/forms.py

```diff
@@ -140,6 +141,24 @@ class Meta:
                                      mce_attrs={"theme" : "advanced", "cleanup_on_startup" : True, "theme_advanced_toolbar_location" : "top", "gecko_spellcheck" : True}),
                     'DELETE' : forms.HiddenInput, 'container' : forms.HiddenInput}
 
+    def clean_label(self):
+        cleaned_data = super(PageForm, self).clean()
+        label = cleaned_data.get('label')
+        if label:
+            existing_website = models.Page.objects.filter(
+                competition=self.instance.competition,
+                label=label
+            ).exclude(pk=self.instance.pk)
+            if existing_website.exists():
+                raise forms.ValidationError(
+                    {
+                        'name': ['Invalid value for website name. A website already exists with that name.']
+                    },
+                    code='invalid'
```

Comment:
can we figure out what this should actually be?

Reply:
code='invalid' is proper Validation Handling according to Django. 
  
https://docs.djangoproject.com/en/dev/ref/forms/validation/#raising-validationerror